src/coordination/coordinator_instance.cpp

```diff
@@ -202,12 +202,33 @@ auto CoordinatorInstance::ShowInstances() const -> std::vector<InstanceStatus> {
     return coordinator_id == curr_leader ? "leader" : "follower";
   };
 
-  auto const coord_instance_to_status = [this, &stringify_coord_health, &get_coord_role](
-                                            CoordinatorToCoordinatorConfig const &instance) -> InstanceStatus {
+  auto const get_coordinator_server = [](CoordinatorToCoordinatorConfig const &instance) -> std::string {
+    // If I am the 1st leader, I need separate processing. Only, the 1st leader will have coordinator_server set to
+    // 0.0.0.0. Coordinators that have been registered after the 1st leader will have coordinator_server set to the
+    // actual IP address so for them we can just retrieve their socket address.
+    if (instance.coordinator_server.GetResolvedIPAddress() == "0.0.0.0") {
+      return fmt::format("{}:{}", instance.coordinator_hostname, instance.coordinator_server.GetPort());
+    }
+    return instance.coordinator_server.SocketAddress();
+  };
+
+  auto const get_bolt_server = [](CoordinatorToCoordinatorConfig const &instance) -> std::string {
+    // If I am the 1st leader, I need separate processing. Only, the 1st leader will have bolt_server set to 0.0.0.0.
+    // Coordinators that have been registered after the 1st leader will have bolt_server set to the actual IP address so
+    // for them we can just retrieve their socket address.
+    if (instance.bolt_server.GetResolvedIPAddress() == "0.0.0.0") {
+      return fmt::format("{}:{}", instance.coordinator_hostname, instance.bolt_server.GetPort());
+    }
+    return instance.bolt_server.SocketAddress();
+  };
```

Comment:
You can group these two functions, and provide default value which should be returned in case coordinator is not 1s leader

Reply:
Here is an example:

```
 auto const get_resolved_address = [](auto const &querying_address,  auto const &resolved_address, auto const &default_value) -> std::string {
    // If I am the 1st leader, I need separate processing. Only, the 1st leader will have querying address set to 0.0.0.0.
    // Coordinators that have been registered after the 1st leader will have bolt_server set to the actual IP address so
    // for them we can just retrieve their socket address.
    if (querying_address == "0.0.0.0") {
      return resolved_address
    }
    return default_value;
 };


.coordinator_server = get_coordinator_server(instance.coordinator_server.GetResolvedIPAddress(), fmt::format("{}:{}", instance.coordinator_hostname, instance.bolt_server.GetPort()), instance.coordinator_server.SocketAddress()), 
.bolt_server = ...
```

Reply:
resolved on the call